File: Array/zhanchengguo/0619-删除所有相邻重复项/Main.java
package com.example.kotlinproject;

public class Main {

    void main() {
        Solution solution = new Solution();
        String s = solution.removeDuplicates("abbaca");

    }
}

class Solution {
    public String removeDuplicates(String S) {
        StringBuilder sb = new StringBuilder();
        int currentPos = 0;
        for (int i = 0; i < S.length(); i++) {
            if (currentPos != 0 && S.charAt(i) == sb.charAt(currentPos - 1)) {
                sb.deleteCharAt(currentPos - 1);
                currentPos--;
            } else {
                sb.append(S.charAt(i));
                currentPos++;
            }
        }
        return sb.toString();
    }
}

File: Array/zhanchengguo/0619-删除所有相邻重复项/main.kt
package com.example.kotlinproject

import java.lang.StringBuilder

fun main() {

    var solution = Solution()
    val result = solution.removeDuplicates("abbaca")
}

class Solution {

    fun removeDuplicates(S: String): String {
        var sb = StringBuilder()
        var currentPos = 0
        for (i in 0 until S.length) {
            if (currentPos != 0 && S.get(i) == sb.get(currentPos - 1)) {
                sb.deleteCharAt(currentPos - 1)
                currentPos--
            } else {
                sb.append(S.get(i))
                currentPos++
            }
        }
        return sb.toString();
    }
}